src/FunctionApps/python/IntakePipeline/__init__.py

```diff
@@ -1,46 +1,114 @@
 import logging
 
+from typing import Tuple
+
 import azure.functions as func
 
 from IntakePipeline.transform import transform_bundle
 from IntakePipeline.linkage import add_patient_identifier
 from IntakePipeline.fhir import (
-    read_fhir_bundles,
     upload_bundle_to_fhir_server,
     store_bundle,
+    store_message,
     get_fhirserver_cred_manager,
+    AzureFhirserverCredentialManager,
+)
+from IntakePipeline.conversion import (
+    convert_batch_messages_to_list,
+    convert_message_to_fhir,
 )
 
 from config import get_required_config
 
 from phdi_transforms.geo import get_smartystreets_client
 
 
-def run_pipeline():
+def run_pipeline(
+    bundle: dict,
+    datatype: str,
+    cred_manager: AzureFhirserverCredentialManager,
+):
     salt = get_required_config("HASH_SALT")
     geocoder = get_smartystreets_client(
         get_required_config("SMARTYSTREETS_AUTH_ID"),
         get_required_config("SMARTYSTREETS_AUTH_TOKEN"),
     )
-    fhirserver_cred_manager = get_fhirserver_cred_manager(
-        get_required_config("FHIR_URL")
+
+    transform_bundle(geocoder, bundle)
+    add_patient_identifier(salt, bundle)
+    store_bundle(
+        get_required_config("INTAKE_CONTAINER_URL"),
+        get_required_config("VALID_OUTPUT_CONTAINER_PATH"),
+        bundle,
+        datatype,
+    )
+    upload_bundle_to_fhir_server(cred_manager, bundle)
+
+
+def convert_to_fhir(
+    blob: func.InputStream,
+    fhir_url: str,
+    container_url: str,
+    invalid_output_path: str,
+    cred_manager: AzureFhirserverCredentialManager,
+) -> Tuple[str, dict]:
+    if blob.name[-3:].lower() not in ("hl7", "xml"):
+        raise Exception(f"invalid file extension for {blob.name}")
+
+    filetype = blob.name.split("/")[-2].lower()
+
+    if filetype == "elr":
+        message_type = "oru_r01"
+        message_format = "hl7v2"
+        bundle_type = "ELR"
+    elif filetype == "vxu":
+        message_type = "vxu_v04"
+        message_format = "hl7v2"
+        bundle_type = "VXU"
+    elif filetype == "eICR":
+        message_type = "ccda"
+        message_format = "ccda"
+        bundle_type = "ECR"
+    else:
+        raise Exception(f"Found an unidentified message_format: {filetype}")
+
+    # VA sends \\u000b & \\u001c in real data, ignore for now
+    messages = convert_batch_messages_to_list(
+        blob.read().decode("utf-8", errors="ignore")
     )
 
-    container_url = get_required_config("INTAKE_CONTAINER_URL")
-    container_prefix = get_required_config("INTAKE_CONTAINER_PREFIX")
-    output_path = get_required_config("OUTPUT_CONTAINER_PATH")
+    for i, message in enumerate(messages):
+        token = cred_manager.get_access_token()
```

Comment:
FWIW, the cred manager keeps track of an access token + expiration date.  This call will check the token (which is tracked in memory) isn't about to expire, and only get a new one if it needs to.  So the cost is relatively low to keeping it in the loop.  But Graham has also rightly suggested that it isn't necessary since the function only lives for bundle at a time the token really should never expire.  Note the downside of this is we get a new token for every bundle (or at least every file).  No strong preference if you change this - I think it will work either way.  Just adding some context on how it was designed.